packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -354,6 +355,11 @@ class SnackBar extends StatefulWidget {
   ///       right padding.
   final EdgeInsetsGeometry? padding;
 
+ /// Optional size constraints for the snack bar.
+  ///
+  /// Defaults to null. If [width] is specified, this property will be ignored.
+  final BoxConstraints? constraints;
```

Comment:
This seems like a strangely generic property to add to Snackbar. If you need this, why not just wrap the `content` directly in a `ConstrainedBox`?

Reply:
Because `padding` has a default value.
So if I want the right size, I have to set padding to zero and set`ConstrainedBox` in `content`.

Reply:
Setting padding to zero doesn't seem like the worst thing in the world. I would lean against adding this property. If we do end up adding this we need to expand the documentation to tell people a little more about what it is used for (see also https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#documentation-dartdocs-javadocs-etc), but I am not sure I would bother since there are better ways to achieve this using composition (even if you have to manually disable padding).

Reply:
Okay